torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -327,6 +330,10 @@ def powerSGD_hook(
         bucket (dist.GradBucket): Bucket that stores a 1D flattened gradient tensor that batches multiple per-variable tensors.
             Note that since DDP comm hook only supports single process single device mode,
             only exactly one tensor is stored in this bucket.
+        batch_tensors_with_same_shape (bool): A boolean value that when set to ``True``, compress and decompress tensors with
+            same shape in a batched operation to achieve higher parallelism. Note that you should also increase the bucket size
+            to make more same-shaped tensors appear in the same bucket, however this may reduce the overlap between computation
+            and communication. Set to ``True`` if the compression / decompression computation is a bottleneck.
```

Comment:
Please add "and increase the memory footprint due to stacking the tensors of the same shape" as another potential downside.